hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -80,6 +81,15 @@ public void stop() {
     if (hiveLocalDir != null) {
       hiveLocalDir.delete();
     }
+
+    // remove raw store if exists
+    try {
+      Method cleanupRawStore = HiveMetaStore.class.getDeclaredMethod("cleanupRawStore");
+      cleanupRawStore.setAccessible(true);
+      cleanupRawStore.invoke(null);
+    } catch (Exception e) {
+      // no op
+    }
```

Comment:
It's usually a code smell to need to access a private method like this. I get that these are just tests, but this feels like something that could likely be handled via a specific config (perhaps in hive-conf.xml).

Reply:
Metastore has been never tested/used by stopping and starting in the same process again and again. Hive tests are running in fork mode so this issue has not been faced before. Gradle tests are running in a same process, and changing the database connection URL which fouls up the db connection pool. There is no config for this in Hive 😢

Reply:
The rationale sounds reasonable to me. We have a reflection library, `DynMethods` that should be used instead of using Java's reflection API directly though.

Reply:
Yeah I agree. That rationale seems reasonable to me as well.